homeassistant/components/tplink/entity.py

```diff
@@ -368,6 +379,11 @@ def _entities_for_device[
                     feat, descriptions, device=device, parent=parent
                 )
             )
+            and async_check_create_deprecated(
+                hass,
+                cls._get_feature_unique_id(device, desc),
+                desc,
+            )
```

Comment:
It a bit unexpected that this happens as part of the listcomp to build the list of entities since it has the side effect of possibly creating an issue

Reply:
Moved reply [to comment below](https://github.com/home-assistant/core/pull/126349#issuecomment-2367635291)